Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PostProcessing/PostProcessUtil.azsli

```diff
@@ -8,41 +8,9 @@
 
 #pragma once
 
-/*
```

Comment:
Currently there is real keyword version of Srgb_To_LinearSrgb and LinearSrgb_To_Srgb. Maybe follow the same pattern here and add a **float** one and a **half** one. That way unnecessary up/down casts can be avoided from different shaders on mobile plat.

Reply:
Isn't real3 supposed to take care of those issues?

Reply:
Yeah but not al the shaders are converted. So if this function is called from a shader like the UI one then on mobile you are up/down casting for just one ALU instruction and is not worth it. Having two overloaded functions avoid this issue as the compiler will pick the correct one based on type.

Reply:
Got it. Added the `float3 LinearSrgb_To_SrgbF(float3 color)` and `float3 Srgb_To_LinearSrgbF(float3 color)` and
`ImGuiAtom.azsl` and `LyShineUI.azsl` now call those functions.